### PR TITLE
Layout: Replace `LayoutLoggedOut` and `LayoutLoggedOutOAuth` with `Layout`

### LINKED ISSUE #2960: Layout: Use a single layout component
Eventually, the following will be true in Calypso:

- It will be possible to sign up from within the desktop app.
- Users will be logged in at the end of the signup flow without being redirected through `wp-login.php`.

Right now, we use separate layout components for when the user is logged in/out or on the login screen. To make it possible to transition between these states without wiping out `#content`, we should use a single layout component.

Much of the work for this is already exists in #2813.

cc @scruffian

### PR DESCRIPTION
**Pull request by drewblaisdell**
Fixes #2960. The layout changes based on the current section and presence of a `user` prop.

`LayoutLoggedOutDesign` hasn't been removed yet because it is also rendered on the server and has dependencies (like `abtest`) that assume a browser environment. I started working on this, but updating `Layout` to run on the server is a rather large task, and will be handled separately - I'll open an issue for this once this is merged.

**Testing**
There are no visual changes. We need to assert that the layout doesn't break when the user is logged out, logged in, using the desktop app, and on `/design`.

*Logged out*
- Visit `/start` while logged out, and assert that the page isn't broken.
- Go through the signup flow an ensure the email verification notice shows.

*Logged in*
- Visit any page while logged in, and assert that the page isn't broken.

*Using the desktop app*
- Check out this branch inside the `calypso/` directory in the desktop app (`cd calypso && git checkout update/one-layout`).
- `make run` from `wp-desktop/` and assert that the login page isn't broken.
- Log in and assert that the Reader (or any other page, once logged in) isn't broken.

*`/design`*
- Visit `/design` while logged out and assert that the page isn't broken.

- [x] Code review
- [x] Product review

### CODE REVIEW

File: client/auth/style.scss [hunk 1]
```diff
@@ -2,22 +2,26 @@
...
-		.wp-content {
-			margin: 0 auto;
-			overflow: hidden;
-			max-width: none;
-			padding-left: 0;
-			padding-right: 0;
-		}
+.is-section-login {
+	background: $blue-wordpress;
+	height: 100%;
```
Comment (drewblaisdell) [inline client/auth/style.scss]: This block obviates [most of these](https://github.com/Automattic/wp-desktop/blob/master/public_desktop/wordpress-desktop.css#L134) styles. They should be removed once this is merged.

File: client/layout/index.jsx [hunk 2]
```diff
@@ -81,7 +82,23 @@ Layout = React.createClass( {
 		return sortBy( this.props.sites.get(), property( 'ID' ) ).pop();
 	},
 
-	renderMasterbar() {
+	renderEmailVerificationNotice: function() {
```
Comment (scruffian) [inline client/layout/index.jsx]: This doesn't seem to be called anywhere anymore.

Comment (scruffian) [inline client/layout/index.jsx]: I added it in eb1059e22ce0c16682238e8ea22ab282541297fc

**Comment by scruffian** [comment]
Code looks good. @drewblaisdell I added a couple of tiny commits.

File: client/layout/index.jsx [hunk 3]
```diff
@@ -98,14 +115,35 @@ Layout = React.createClass( {
...
+		translatorInvitation = ! showWelcome &&
+				translatorInvitation.isPending() &&
+				translatorInvitation.isValidSection( this.props.section );
+
+		return (
+			<span>
+				<Welcome isVisible={ showWelcome } closeAction={ this.closeWelcome } additionalClassName="NuxWelcome">
+					<WelcomeMessage welcomeSite={ newestSite } />
+				</Welcome>
+				<TranslatorInvitation isVisible={ showInvitation } />
```
Comment (stephanethomas) [inline client/layout/index.jsx]: `showInvitation` is `undefined`.

File: client/layout/index.jsx [hunk 4]
```diff
@@ -98,14 +115,35 @@ Layout = React.createClass( {
...
+			showWelcome,
+			newestSite;
+
+		if ( ! this.props.user ) {
+			return null;
+		}
+
+		showWelcome = this.props.nuxWelcome.getWelcome();
+		newestSite = this.newestSite();
+		translatorInvitation = ! showWelcome &&
```
Comment (stephanethomas) [inline client/layout/index.jsx]: `translatorInvitation` is never used.

File: client/layout/index.jsx (hunk 3 above)
Comment (drewblaisdell) [inline client/layout/index.jsx]: Fixed. Thanks for catching that, @stephanethomas.

**Comment by scruffian** [comment]
Tested again on desktop, logged in and logged out. LGTM.

**Comment by scruffian** [comment]
> LayoutLoggedOutDesign hasn't been removed yet because it is also rendered on the server and has dependencies (like abtest) that assume a browser environment. I started working on this, but updating Layout to run on the server is a rather large task, and will be handled separately - I'll open an issue for this once this is merged.

I did this in #3112.

File: client/auth/style.scss (hunk 1 above)
Comment (scruffian) [inline client/auth/style.scss]: > This block obviates most of these styles. They should be removed once this is merged.

I created an issue for that here: https://github.com/Automattic/wp-desktop/issues/67

**Pull request merged**